src/RtspConnection.c

```diff
@@ -579,13 +579,20 @@ int performRtspHandshake(void) {
         }
 
         sessionId = getOptionContent(response.options, "Session");
-        if (sessionId == NULL) {
+
+        // Strips session id in the case the 
+        // RTSP server returns excess data.
+        // (i.e: DEADBEEF;timeout=)
+        char* session;
+        session = strtok(sessionId, ";");
```

Comment:
Check for `sessionId == NULL` first. We can't be sure someone hasn't called `strtok()` before and not read all available tokens.

Reply:
This is my first pull request - I have committed updates, I think I am doing this properly.

Follow up comments of the changes are below.